doc/adding_to_an_adapter.md

```diff
@@ -1,120 +1,241 @@
 Adding to an Adapter
 ====================
-## Defining an Adapter
-To define an adapter, use the top-level `AdapterKind` element in the `conf/describe.xml` file. Only one adapter can be defined in the `describe.xml` file. The key will be used when creating objects (See [Creating an object](#creating-an-object)), and must also be present in the `manifest.txt` file in the `"adapter_kinds"` array. 
 
-> Note: The `mp-init` tool will create the AdapterKind element and ensure its key exists in the `manifest.txt` file.
+## Object Model
+
+### What is the Object Model?
+At runtime, Aria Operations Manager creates objects representing data sources in the adapter. Every adapter must include
+an object model that provides object types, object properties, and semantic definitions describing resources and the
+metric data's meaning. The object model can be defined by using a `describe.xml` file, which should be defined in the
+`conf` directory, or it can be defined using the [definition module](../lib/python/doc/aria/ops/definition.html)
+from the [VMware Aria Operations Integration SDK Library](https://pypi.org/project/vmware-aria-operations-integration-sdk-lib/)
+by defining the [AdapterDefinition](../lib/python/doc/aria/ops/definition/adapter_definition.html) object returned by
+The **get_adapter_definition()** method in `app/adapter.py`.
+
+### describe.xml Vs. AdapterDefinition?
+when running `mp-build`, `mp-test collect `, or `mp-test test` they call the **get_adapter_definition** method
+and then generate `describe.xml` file with the [AdapterDefinition](../lib/python/doc/aria/ops/definition/adapter_definition.html)
+object. If a `describe.xml` file is present in the `conf` directory, then the object returned by the **get_adapter_definition**
+method will be ignored.
+
+## Defining an Adapter and Adapter Instance in the Object Model
+To define an adapter in the `conf/describe.xml` file use the top-level `AdapterKind` element. Only one adapter can be
+defined in the [object model](#defining-an-adapter-and-adapter-instance-in-the-object-model). The `key` will be used when creating objects (See [Creating an object](#creating-an-object)),
+and must also be present in the `manifest.txt` file in the `"adapter_kinds"` array. When defining an adapter, we also
+have to define an adapter instance. An adapter instance is a special object in VMware Aria Operations that stores user
+configuration for a connection. Every adapter must have exactly one adapter instance type. Adapter instances are set by
+defining a `ResourceKind` element with attribute `type=7`. in Python,
+we can use the [AdapterDefinition](../lib/python/doc/aria/ops/definition/adapter_definition.html).
```

Comment:
I think there's still some places we are saying 'adapter instance' where we mean 'adapter instance type'

Reply:
"An adapter instance type is a special object" should go back to just saying "An adapter instance is a special object" - we need to make sure we're distinguishing between what is an object (resource) in Aria Ops, vs an class/type of object (resourceKind).

The other changes here look correct.